packages/hierarchies-react/src/presentation-hierarchies-react/itwinui/TreeErrorRenderer.tsx

```diff
@@ -3,61 +3,105 @@
       return (
-        <ResultSetTooLarge
+        <ErrorItemContainer
           key={errorNode.error.id}
-          onFilterClick={() => {
-            const hierarchyLevelDetails = getHierarchyLevelDetails?.(errorNode.parent?.id);
-            hierarchyLevelDetails && onFilterClick?.(hierarchyLevelDetails);
-          }}
-          onOverrideLimit={getHierarchyLevelDetails ? (limit) => getHierarchyLevelDetails(errorNode.parent?.id)?.setSizeLimit(limit) : undefined}
-          scrollToElement={scrollToElement}
-          limit={errorNode.error.resultSetSizeLimit}
           errorNode={errorNode}
+          actions={[
+            {
+              action: () => {
+                onOverrideLimit?.();
+              },
+              label: localizedStrings.increaseHierarchyLimit.replace("{{limit}}", MAX_LIMIT_OVERRIDE.toString()),
+              condition: () => !!onOverrideLimit && limit < MAX_LIMIT_OVERRIDE,
+            },
+            {
+              action: () => {
+                const hierarchyLevelDetails = getHierarchyLevelDetails?.(errorNode.parent?.id);
+                hierarchyLevelDetails && onFilterClick?.(hierarchyLevelDetails);
+              },
+              label: localizedStrings.increaseHierarchyLimitWithFiltering,
+              condition: () => !!onFilterClick && !!errorNode.parent?.isFilterable,
+            },
+          ]}
+          message={localizedStrings.resultLimitExceeded.replace("{{limit}}", limit.toString())}
+          scrollToElement={() => scrollToElement(errorNode)}
         />
       );
     }
     if (errorNode.error.type === "NoFilterMatches") {
       return (
-        <NoFilterMatches
+        <ErrorItemContainer
           key={errorNode.error.id}
-          scrollToElement={scrollToElement}
-          onFilterClick={() => {
-            const hierarchyLevelDetails = getHierarchyLevelDetails?.(errorNode.parent?.id);
-            hierarchyLevelDetails && onFilterClick?.(hierarchyLevelDetails);
-          }}
           errorNode={errorNode}
+          actions={[
+            {
+              action: () => {
+                const hierarchyLevelDetails = getHierarchyLevelDetails?.(errorNode.parent?.id);
+                hierarchyLevelDetails && onFilterClick?.(hierarchyLevelDetails);
+              },
+              label: localizedStrings.noFilteredChildrenChangeFilter,
+              condition: () => true,
+            },
+          ]}
+          message={localizedStrings.noFilteredChildren}
+          scrollToElement={() => scrollToElement(errorNode)}
         />
       );
     }
-    return <DefaultErrorContainer key={errorNode.error.id} scrollToElement={scrollToElement} reloadTree={reloadTree} errorNode={errorNode} />;
+    return (
+      <ErrorItemContainer
+        key={errorNode.error.id}
+        errorNode={errorNode}
+        actions={[
+          {
+            action: () => reloadTree?.({ parentNodeId: errorNode.parent?.id, state: "reset" }),
```

Comment:
@grigasp @saskliutas In the new error workflow failing to create a hierarchy displays retry button in errorContainer and retry action near the node, but the reload action itself might not be passed which could cause confusion on how to reload the tree. Should we make it required? Ps. you can still reload the tree with closing the expander and reopening it.

Reply:
I wouldn't mind it being required, especially since `useTree` always returns this function.

Reply:
Made it required.